tests/test_context.py

```diff
@@ -48,3 +50,17 @@ def test_message_handler(device):
     assert context.message_handler is message_handler_test
     with context:
         assert current_context().message_handler is context.message_handler
+
+
+def test_async_wake_interval(device):
+    """Test read-write property async_wake_interval."""
+    with Context(device) as context:
+        context.async_wake_interval = 42
+        assert isclose(context.async_wake_interval, 42)
+
+
+def test_droppler_factor(device):
+    """Test write property doppler_factor."""
+    with Context(device) as context, Source(context) as source:
+        context.doppler_factor = 4/9
+        assert isclose(source.doppler_factor, 4/9)
```

Comment:
I don't think setting the context doppler factor will change the source's one, but the actual effect that we listen to, i.e. output doppler = source doppler *times* context doppler.

As mentioned in https://github.com/McSinyx/palace/issues/48#issue-576938127, I'm not sure how `Context.doppler_factor` can be tested further than just setting the value.  However, you can be a little more exhaustive by additionally supplying negative and large numbers also.